### PR TITLE
feat: added pipelines for building docker images

### PR DESCRIPTION
**Pull request by TwelveNights**
Some progress for #32

### CODE REVIEW

File: .github/workflows/build-docker-api.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.5.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Increase watcher limit
```
Comment (GabrielBottari) [inline .github/workflows/build-docker-api.yml:21]: I don't believe you need this. this was for the webapp

File: .github/workflows/build-docker-webapp.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.5.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Increase watcher limit
```
Comment (GabrielBottari) [inline .github/workflows/build-docker-webapp.yml:21]: Also i'm not sure if it was for testing or build, so might be useless too

File: .github/workflows/build-docker-api.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        env:
+          nx_cloud_token: ${{ secrets.NX_CLOUD_TOKEN }}
+        run:
+          sed -i "s/nx_cloud_token/$nx_cloud_token/" $GITHUB_WORKSPACE/nx.json
+
+      - name: Install npm packages
+        run: npm ci
+
+      - name: Build api
+        run: npx nx build api -- --base=origin/master
```
Comment (GabrielBottari) [inline .github/workflows/build-docker-api.yml:43]: you're already installing npm packages, therefore you can use npm run build api no?  no need for npx. also, the double `--` is for passing down arguments when using npm

File: .github/workflows/build-docker-webapp.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        env:
+          nx_cloud_token: ${{ secrets.NX_CLOUD_TOKEN }}
+        run:
+          sed -i "s/nx_cloud_token/$nx_cloud_token/" $GITHUB_WORKSPACE/nx.json
+
+      - name: Install npm packages
+        run: npm ci
+
+      - name: Build webapp
+        run: npx nx build webapp -- --base=origin/master
```
Comment (GabrielBottari) [inline .github/workflows/build-docker-webapp.yml:43]: same comment as above

File: dockerfiles/api/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM node:alpine
+WORKDIR /opt
+COPY package.json package-lock.json dist/apps/api ./
```
Comment (GabrielBottari) [inline dockerfiles/api/Dockerfile:3]: you're copying the whole repo with ./ ?

**Review by GabrielBottari — COMMENTED** [review]
Good stuff, so you're going to dispatch those actions from the main CI? Could you potentially pass the dist folder as parameter so you don't have to build it twice?

Also, with `nx affected:apps --base=master --plain`, you can print the apps affected by change, and you could use if conditions to only dispatch those workflows, and not redeploy for no reason

File: .github/workflows/build-docker-api.yml (hunk 3 above)
Comment (GabrielBottari) [reply .github/workflows/build-docker-api.yml:43]: also the base option is unused here. its only used with affected commands to know where to compare from, so you can remove them in both workflows

File: dockerfiles/api/Dockerfile (hunk 5 above)
Comment (TwelveNights) [reply dockerfiles/api/Dockerfile:3]: No, I'm copying the preceding files to the current working directory on the docker image.

File: .github/workflows/build-docker-api.yml (hunk 3 above)
Comment (TwelveNights) [reply .github/workflows/build-docker-api.yml:43]: thanks! I didn't know that

**Pull request closed without merging**